parser.py

```diff
@@ -1 +1,40 @@
-# TODO Xin - Add methods here that we will need. Try to think carefully about organization and how to separate functionality effectively.
\ No newline at end of file
+# TODO Xin - Add methods here that we will need. Try to think carefully about organization and how to separate functionality effectively.
+from bs4 import BeautifulSoup as bs
+from bs4 import SoupStrainer as ss
+import re
+import urllib2 as ul
+ 	
+def links(webpage):
+	links = []
+	biglist = ul.urlopen(webpage)
+	soup = bs(biglist, "html.parser")
+	tablebody = soup.body.find(id="content")
+	bodycontent = tablebody.find(id="bodyContent")
+	table = bodycontent.find(id="mw-content-text").table
+	for n in range(1, 505):
+		row = table.find_all('tr')[n]
```

Comment:
Pull the "find_all" out of the for loop. Otherwise you're finding all 505 times instead of once!
```python
rows = table.find_all('tr')
for row in rows():
    name = row.find_all('td')[1]
    # etc
```